src/pages/Search/Search.js

```diff
@@ -1,23 +1,76 @@
-import React from 'react';
-import SearchBooks from './SearchBooks';
-import Categories from './Categories';
-import Suggestions from './Suggestions';
+import React, { useState } from 'react';
 import styled from 'styled-components';
+import SearchMain from './SearchMain';
+import SearchFilter from './SearchFilter';
 
 const Search = () => {
+  const [{ type, placeholder, api }, setFilter] = useState({
+    type: '전체',
+    placeholder: '검색어를 입력하세요',
+    api: 'all',
+  });
+
+  const [isOnFocus, setIsOnFocus] = useState(false);
+  const isInputFocusOn = () => {
+    setIsOnFocus(true);
+  };
+
+  const isInputFocusOut = () => {
+    setIsOnFocus(false);
+  };
```

Comment:
```suggestion
  const inputFocusOn = () => {
    setIsOnFocus(true);
  };

  const inputFocusOut = () => {
    setIsOnFocus(false);
  };
```
- `is-` 네이밍은 boolean 타입에 적합해보여요
- 함수가 너무 많이 만들어지는거 같으면 일단 setIsOnFocus만 내려주고, 해당 컴포넌트에서 필요한 동작을 정의하도록 해도 좋을거 같아요